src/content/services/monetizationTagManager.ts

```diff
@@ -294,7 +323,10 @@ export class MonetizationTagManager extends EventEmitter {
 
     this.monetizationTags.set(tag, details)
 
-    if (walletAddress) startMonetization({ requestId, walletAddress })
+    if (walletAddress) {
+      startMonetization({ requestId, walletAddress })
+      isTabMonetized({ value: true })
+    }
```

Comment:
We should bulk send (for start, resumse, stop) all the link tags that are in the page. This way we can avoid sending multiple events. With the current approach we send two `START_MONETIZATION` and two `TAB_MONETIZED` when a new link tag is added.